clients/venice-client/src/main/java/com/linkedin/venice/fastclient/stats/ClusterRouteStats.java

```diff
@@ -0,0 +1,115 @@
+package com.linkedin.venice.fastclient.stats;
+
+import com.linkedin.venice.read.RequestType;
+import com.linkedin.venice.stats.AbstractVeniceHttpStats;
+import com.linkedin.venice.stats.StatsUtils;
+import com.linkedin.venice.stats.TehutiUtils;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.OccurrenceRate;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ClusterRouteStats {
+  private static final Logger LOGGER = LogManager.getLogger(ClusterRouteStats.class);
+
+  private static final ClusterRouteStats DEFAULT = new ClusterRouteStats();
+
+  private final Map<String, RouteStats> perRouteStatMap = new VeniceConcurrentHashMap<>();
+
+  public static ClusterRouteStats get() {
+    return DEFAULT;
+  }
+
+  private ClusterRouteStats() {
+  }
+
+  public RouteStats getRouteStats(
+      MetricsRepository metricsRepository,
+      String clusterName,
+      String instanceUrl,
+      RequestType requestType) {
+    String combinedKey = clusterName + "-" + instanceUrl + "-" + requestType.toString();
+    return perRouteStatMap.computeIfAbsent(combinedKey, ignored -> {
+      String instanceName = instanceUrl;
+      try {
+        URL url = new URL(instanceUrl);
+        instanceName = url.getHost() + "_" + url.getPort();
```

Comment:
What's the reason to include port here? Asking because it might affect our expected behavior when we start using gRPC which have different port.

Reply:
`Port` is actually not required from my understanding considering Venice Server is not sharing the hardware with other components and I can remove it.